template_reports/templating/permissions.py

```diff
@@ -22,32 +23,32 @@ def has_view_permission(obj, perm_user):
     return True
 
 
-def _check_permissions(item, perm_user):
+def _check_permissions(item, check_permissions: Callable[[object], bool]):
```

Comment:
remove _check_permissions it's now redundant, you can simplify by moving the exception directly into enforce_permissions. you'll need to change the patch inside test_integration_permission_denied too.